File: plusOne.py
class Solution:
    def (self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        i = len(digits) - 1
        while digits[i] + 1 > 9:
            digits[i] = 0
            if i == 0:
                digits.insert(0, 1)
                return digits
            i -= 1
        digits[i] += 1
        return digits
            
        
        
        """
        Original Solution 
        s = ""
        ans = []
        for d in digits:
            s += str(d)
        num = str(int(s) + 1)
        for n in num:
            ans.append(int(n))
        return ans
        */
        """

File: Cpp/Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int N = digits.size();
        vector<int> result = digits;
        if (N == 0) return result;
        int carry = 1;
        for (int i = N - 1; i > -1; i--) {
            if (carry == 0) break;
            carry = 0;
            int added = digits[i] + 1;
            if (added > 9) {
                added -= 10;
                carry = 1;
            }
            result[i] = added;
        }
        if (carry == 1) {
            result.insert(result.begin(), 1);
        }
        return result;
    }
};